UM-USElab/gradecraft-development
# Angularize dashboard due this week component

### Status
READY

### Description
This PR Angularizes the due this week component that is displayed on the dashboard. After investigating via the use of `rack-mini-profiler`, it was found that this component takes the longest to render.

Step one is to load this component asynchronously and step two is to see if we can get rid of the presenter and maybe improve upon some of the back-end querying.

### Migrations
NO

### Steps to Test or Reproduce
This one is a difficult one to test. This widget obviously displays assignments that have been predicted by the student as well as those that are due within the week.

The view also differs for an instructor versus a student. We should also ensure that with the submission button refactor, there is not an errant scenario where the button is showing when it isn't supposed to or it is showing the wrong text or links to the wrong page.

### Impacted Areas in Application
![screen shot 2018-03-15 at 11 09 54 am](https://user-images.githubusercontent.com/17604722/37472025-6f14705e-2841-11e8-942a-24b00cfb5e2e.png)

======================
Relates to #3880
